.github/workflows/run-test-suite.yml

```diff
@@ -0,0 +1,89 @@
+name: run-test-suite
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  test:
+    services:
+      postgres:
+        image: postgres:10.8
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_DB: test
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+      mysql:
+        image: mysql:5.7
+        env:
+          MYSQL_ROOT_PASSWORD: password
+        ports:
+          - 3306
+        options: >-
+          --health-cmd="mysqladmin ping"
+          --health-interval=10s
+          --health-timeout=5s
+          --health-retries=3
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        include:
+          - ruby: 2.7
+            activerecord: 52
+          - ruby: 2.7
+            activerecord: 60
+          - ruby: 2.7
+            activerecord: 61
+          - ruby: 3.0
+            activerecord: 61
+    env:
+       PGUSER: postgres
+       PGPASSWORD: postgres
+       PGDATABASE: postgres
+       PGHOST: 127.0.0.1
+       PGPORT: 5432
+       MYSQL_PWD: password
+       MYSQL_HOST: 127.0.0.1
+       BUNDLE_GEMFILE: gemfiles/gemfile_${{ matrix.activerecord }}.gemfile
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Setup Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+          bundler-cache: true # runs 'bundle install' and caches installed gems automatically
+      - name: Install gems
+        run: |
+          bundle config path vendor/bundle
+          bundle install --jobs 4 --retry 3
+
+      - name: Setup test database
+        run: |
+          sudo /etc/init.d/mysql start
+          mysql --port ${{ job.services.mysql.ports['3306'] }} -uroot -ppassword -e 'create database IF NOT EXISTS hash_options_test;'
```

Comment:
did the mysql environment variables help us here or elsewhere?
maybe getting rid of those would just cut down on the gook in this file

Reply:
I can't, I can't take any of this out, I'm sorry, it will not work without this gook or the other gook in the env section above, it needs all the gook. I've checked so many times. It needs the gook.